lib/pushmi_pullyu/aip_downloader_files_directories.rb

```diff
@@ -0,0 +1,91 @@
+require 'rdf'
+require 'rdf/n3'
+require 'fileutils'
+
+# Download all of the metadata/datastreams and associated data
+# related to an object
+
+module PushmiPullyu::AipDownloaderFilesDirectories
+  FIXITY_PATH = '/content/fcr:fixity'.freeze
+  CONTENT_DATASTREAM_METADATA_PATH = '/content/fcr:metadata'.freeze
+  CHARACTERIZATION_PATH = '/characterization'.freeze
+  VERSIONS_PATH = '/content/fcr:versions'.freeze
+  CONTENT_PATH = '/content'.freeze
+  THUMBNAIL_PATH = '/thumbnail'.freeze
+  FEDORA3FOXML_PATH = '/fedora3foxml'.freeze
+  FEDORA3FOXML_METADATA_PATH = '/fedora3foxml/fcr:metadata'.freeze
+
+  FILENAME_PREDICATE = 'info:fedora/fedora-system:def/model#downloadFilename'.freeze
+
+  def self.included(_klass)
+    attr_reader :workdir, :basedir, :datadir, :objectsdir, :metadatadir,
+                :logsdir, :thumbnailsdir,
+                :tar_filename, :main_object_filename, :fixity_report_filename,
+                :content_datastream_metadata_filename,
+                :characterization_filename, :versions_filename,
+                :thumbnail_filename, :fedora3foxml_filename,
+                :fedora3foxml_metadata_filename
+  end
+
+  def make_object_directories
+    clean_directories
+    logger.debug("#{noid}: Creating directories ...")
+    FileUtils.mkdir_p(basedir)
+    FileUtils.mkdir_p(datadir)
+    FileUtils.mkdir_p(objectsdir)
+    FileUtils.mkdir_p(metadatadir)
+    FileUtils.mkdir_p(logsdir)
+    FileUtils.mkdir_p(thumbnailsdir)
+    logger.debug("#{noid}: Creating directories done")
+  end
+
+  def clean_directories
+    return unless File.exist?(basedir)
+    logger.debug("#{noid}: Nuking directories ...")
+    FileUtils.rm_rf(basedir)
```

Comment:
I agree with a lot of this feedback. mixin module is leaving me very confused. (especially when its only being mixed/included in one place). Makes code very hard to trace and follow. (things are being instantized in a different class etc., things are just being called, etc.). If the main class is getting to big, maybe extract some of this directory functionality into a directory class for cleaning/making directories? Or look at some of matt's feedback.